cli/src/cli_util.rs

```diff
@@ -1597,6 +1598,29 @@ to the current parents may contain changes from multiple commits.
         ui: &Ui,
         revision_args: &[RevisionArg],
     ) -> Result<IndexSet<Commit>, CommandError> {
+        let all_commits = self
+            .resolve_some_revsets_commit_ids_default_single(ui, revision_args)?
+            .iter()
+            .map(|id| {
+                self.repo()
+                    .store()
+                    .get_commit(id)
+                    .map_err(RevsetEvaluationError::StoreError)
+            })
+            .try_collect()?;
+        Ok(all_commits)
+    }
+
+    /// Evaluates revset expressions to non-empty set of commit IDs. The
+    /// returned set preserves the order of the input expressions.
+    ///
+    /// If an input expression is prefixed with `all:`, it may be evaluated to
+    /// any number of revisions (including 0.)
+    pub fn resolve_some_revsets_commit_ids_default_single(
+        &self,
+        ui: &Ui,
+        revision_args: &[RevisionArg],
+    ) -> Result<IndexSet<CommitId>, CommandError> {
```

Comment:
Maybe better to replace `resolve_some_revsets_default_single()` to return `CommitId`s?

Some callers do need `Commit` objects, but they often use `.iter().ids().collect()`, so I don't think we need two variants of the "default_single" function.